plugins/lookup/lookup_graphql.py

```diff
@@ -0,0 +1,192 @@
+            description:
+                - The API token created through Nautobot
+            env:
+                # in order of precedence
+                - name: NAUTOBOT_TOKEN
+            required: False
+        url:
+            description:
+                - The URL to the Nautobot instance to query (http://nautobot.example.com:8000)
+            env:
+                # in order of precedence
+                - name: NAUTOBOT_URL
+            required: True
+        validate_certs:
+            description:
+                - Whether or not to validate SSL of the Nautobot instance
+            required: False
+            default: True
+        variables:
+            description:
+                - Dictionary of keys/values to pass into the GraphQL query, see [pynautobot GraphQL documentation](https://pynautobot.readthedocs.io/en/latest/advanced/graphql.html) for more details
+            required: False
+    requirements:
+        - pynautobot
+"""
+
+EXAMPLES = """
+  # Make API Query without variables
+  - name: SET FACT OF STRING
+    set_fact:
+      query_string: |
+        query {
+          sites {
+            id
+            name
+            region {
+              name
+            }
+          }
+        }
+
+  # Make query to GraphQL Endpoint
+  - name: Obtain list of sites from Nautobot
+    set_fact:
+      query_response: "{{ query('networktocode.nautobot.lookup_graphql', query=query, url='https://nautobot.example.com', token='<redact>') }}"
+
+  # Example with variables
+  - name: SET FACTS TO SEND TO GRAPHQL ENDPOINT
+    set_fact:
+      variables:
+        site_name: den
+      query_string: |
+        query ($site_name:String!) {
+            sites (name: $site_name) {
+            id
+            name
+            region {
+                name
+            }
+            }
+        }
+
+  # Get Response with variables
+  - name: Obtain list of devices from Nautobot
+    set_fact:
+      query_response: "{{ query('networktocode.nautobot.lookup_graphql', query=query, variables=variables url='https://nautobot.example.com', token='<redact>') }}"
+"""
+
+RETURN = """
+  data:
+    description:
+      - Data result from the GraphQL endpoint
+    type: dict
+"""
+
+
+def nautobot_lookup_graphql(**kwargs):
+    """Lookup functionality, broken out to assist with testing
+
+    Returns:
+        [type]: [description]
+    """
+    # Setup API Token information, URL, and SSL verification
+    url = kwargs.get("url") or os.getenv("NAUTOBOT_URL")
+
+    # Verify URL is passed in, that it is not None
+    if url is None:
+        raise AnsibleLookupError("Missing URL of Nautobot")
+
+    token = kwargs.get("token") or os.getenv("NAUTOBOT_TOKEN")
+    ssl_verify = kwargs.get("validate_certs", True)
+
+    if not isinstance(ssl_verify, bool):
+        raise AnsibleLookupError("validate_certs must be a boolean")
+
+    nautobot_api = NautobotApiBase(token=token, url=url, ssl_verify=ssl_verify)
+    query = kwargs.get("query")
```

Comment:
Should we add some `display.v` to this to show the URL/query string, etc?

Reply:
Makes sense. I'll get that added in!